plugin/kubernetes/metrics.go

```diff
@@ -0,0 +1,77 @@
+package kubernetes
+
+import (
+	"context"
+	"net/url"
+	"time"
+
+	"github.com/coredns/coredns/plugin"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"k8s.io/client-go/tools/metrics"
+)
+
+var (
+	// requestLatency is a Prometheus Summary metric type partitioned by
+	// "verb" and "url" labels. It is used for the rest client latency metrics.
+	requestLatency = promauto.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: plugin.Namespace,
+			Subsystem: "kubernetes",
+			Name:      "rest_client_request_duration_seconds",
+			Help:      "Request latency in seconds. Broken down by verb and URL.",
+			Buckets:   prometheus.ExponentialBuckets(0.001, 2, 20),
```

Comment:
I think that using `prometheus.DefBuckets` would be better, or why do you think that these custom buckets are better?

Reply:
I was not aware of default buckets from prometheus. May we can start with the same & see in future if we require to capture additional buckets.